runtime/syntax/python.vim

```diff
@@ -159,9 +159,9 @@ syn match   pythonEscape	"\\$"
 if !exists("python_no_number_highlight")
   " numbers (including longs and complex)
   syn match   pythonNumber	"\<0[oO]\=\o\+[Ll]\=\>"
-  syn match   pythonNumber	"\<0[xX]\x\+[Ll]\=\>"
+  syn match   pythonNumber	"\<0[xX][0-9A-F_]\+[Ll]\=\>"
```

Comment:
As mentioned in #7923 I'm already preparing a lot of other changes including this one.
The support for Python 2 has already been dropped for keywords and builtins in PR #7855 and I am dropping it in my changes for this part as well.
If you don't mind I'd prefer to send my change in and will try my best to extract it from other changes today and send a PR.